sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -121,24 +121,40 @@ class AstBuilder extends DataTypeAstBuilder with SQLConfHelper with Logging {
       visit(s).asInstanceOf[CompoundBody]
     }.getOrElse {
       val logicalPlan = visitSingleStatement(ctx.singleStatement())
-      CompoundBody(Seq(SingleStatement(parsedPlan = logicalPlan)))
+      CompoundBody(Seq(SingleStatement(parsedPlan = logicalPlan)),
+        java.util.UUID.randomUUID.toString)
     }
   }
 
   override def visitSingleCompoundStatement(ctx: SingleCompoundStatementContext): CompoundBody = {
     visit(ctx.beginEndCompoundBlock()).asInstanceOf[CompoundBody]
   }
 
-  private def visitCompoundBodyImpl(ctx: CompoundBodyContext): CompoundBody = {
+  private def visitCompoundBodyImpl(ctx: CompoundBodyContext, label: String = ""): CompoundBody = {
```

Comment:
What is the reason for having default for label as empty string?
My understanding is that we are always going to provide a value to label (UUID if not provided).

Reply:
```
beginEndCompoundBlock
    : beginLabel? BEGIN compoundBody END endLabel?
    ;
```

For example, `CompoundBody` can be found inside `if/else` or `while` loop and we don't need label for that body because loop will have label.

Reply:
I see. Maybe it is cleaner to use `Option` then to default to empty string?